doc/KUKSA.val_data_broker/authorization_alt_d.md

```diff
@@ -0,0 +1,315 @@
+
+This specification defines a profile for issuing OAuth 2.0 access tokens in JSON Web Token (JWT) format. Authorization servers and resource servers from different vendors can leverage this profile to issue and consume access tokens in an interoperable manner.
+
+The original OAuth 2.0 Authorization Framework [RFC6749] specification does not mandate any specific format for access tokens. While that remains perfectly appropriate for many important scenarios, in-market use has shown that many commercial OAuth 2.0 implementations elected to issue access tokens using a format that can be parsed and validated by resource servers directly, without further authorization server involvement.
+
+This specification aims to provide a standardized and interoperable profile as an alternative to the proprietary JWT access token layouts going forward. Besides defining a common set of mandatory and optional claims, the profile provides clear indications on how authorization request parameters determine the content of the issued JWT access token, how an authorization server can publish metadata relevant to the JWT access tokens it issues, and how a resource server should validate incoming JWT access tokens.
+
+[...]
+
+Authorization Claims [[RFC 9068](https://datatracker.ietf.org/doc/html/rfc9068#name-authorization-claims)]
+
+If an authorization request includes a scope parameter, the corresponding issued JWT access token SHOULD include a "scope" claim as defined in Section 4.2 of [RFC8693].
+All the individual scope strings in the "scope" claim MUST have meaning for the resources indicated in the "aud" claim. See Section 5 for more considerations about the relationship between scope strings and resources indicated by the "aud" claim.
+
+
+## Authorization in kuksa.val
+
+Authorization is achieved by supplying an authorization header as part of any http2 / gRPC request.
+
+`Authorization: Bearer TOKEN`
+
+The bearer token is encoded as a JWT access token.
+
+> **JWT access token**:<br>
+    An OAuth 2.0 access token encoded in JWT format and complying with the requirements described in this specification. [[RFC 9068](https://datatracker.ietf.org/doc/html/rfc9068#section-1.2)]
+
+This token contains everything the server needs to verify the authenticity of the claims
+contained therein. The structure is specified in the RFC as:
+
+#### JWT Access Token - Data Structure
+
+| Claim     | Description                                          |
+|-----------|------------------------------------------------------|
+| iss       | REQUIRED - as defined in Section 4.1.1 of [RFC7519]. |
+| exp       | REQUIRED - as defined in Section 4.1.4 of [RFC7519]. |
+| aud       | REQUIRED - as defined in Section 4.1.3 of [RFC7519]. See Section 3 for indications on how an authorization server should determine the value of "aud" depending on the request. |
```

Comment:
Maybe need to clarify whether this is about _our_ usage, or a recap of JWT in general. For example, while we shoud requrie defined "aud"  claims for databroker, if i read correctly in RFC7519 it is optional

Reply:
Yes, definitely. It's a bit unclear, but the whole "Access token" section (now renamed JWT access token) describes how [RFC 9068](https://datatracker.ietf.org/doc/html/rfc9068) defines a JWT access token.

Or put another way.
* There is one RFC defining JWT, [RFC 7519](https://www.rfc-editor.org/rfc/rfc7519).
* And there's another, newer one, defining JWT **access tokens**, or strictly speaking *JWT Profile for OAuth 2.0 Access Tokens*, [RFC 9068](https://datatracker.ietf.org/doc/html/rfc9068).

My proposal, I guess, is that it would be beneficial to follow, or at least be compatible with [RFC 9068](https://datatracker.ietf.org/doc/html/rfc9068).